AppKit/CPWindow/CPWindow.j

```diff
@@ -1958,6 +1967,79 @@ CPTexturedBackgroundWindowMask
+                        var aTrackingArea   = [trackingAreas objectAtIndex:j],
+                        trackingOptions = [aTrackingArea options];
+                        
+                        if (trackingOptions & CPTrackingEnabledDuringMouseDrag)
+                            
+                            if ((trackingOptions & CPTrackingActiveAlways) ||
+                                ((trackingOptions & CPTrackingActiveInKeyWindow) && ([self isKeyWindow])) ||
+                                ((trackingOptions & CPTrackingActiveWhenFirstResponder) && ([self isKeyWindow]) && (_firstResponder == aView)))
+                                
+                                if (CGRectContainsPoint(((trackingOptions & CPTrackingInVisibleRect) ? [aView visibleRect] : [aTrackingArea rect]), convertedPoint))
+                                {
+                                    [mouseEnteredStack addObject:aTrackingArea];
+                                    
+                                    if (![_mouseEnteredStack containsObject:aTrackingArea])
+                                        
+                                        // Mouse was not in this rect so it's a mouseEntered
+                                        
+                                        if (trackingOptions & CPTrackingMouseEnteredAndExited)
+                                            
+                                            [[aTrackingArea owner] mouseEntered:[CPEvent enterExitEventWithType:CPMouseEntered
+                                                                                                       location:point
+                                                                                                  modifierFlags:[anEvent modifierFlags]
+                                                                                                      timestamp:[anEvent timestamp]
+                                                                                                   windowNumber:_windowNumber
+                                                                                                        context:nil
+                                                                                                    eventNumber:-1
+                                                                                                   trackingArea:aTrackingArea]];
+                                }
+                }
+            }
+            
+            // Search now for exited views (were in _mouseEnteredStack but no more in mouseEnteredStack)
+            
+            for (var i = 0, nb = [_mouseEnteredStack count]; i < nb; i++)
+            {
+                var aTrackingArea   = [_mouseEnteredStack objectAtIndex:i],
+                    trackingOptions = [aTrackingArea options];
+
+                if (trackingOptions & CPTrackingEnabledDuringMouseDrag)
+
+                    if (![mouseEnteredStack containsObject:aTrackingArea])
```

Comment:
Same here, what about dictionary for mouseEnteredStack and _mouseEnteredStack. We could avoid a lot of loop in loop